rad/src/lib.rs

```diff
@@ -201,11 +208,81 @@ pub async fn run_retrieval_report(
 /// Run retrieval stage of a data request, return `RadonTypes`.
 pub async fn run_retrieval(retrieve: &RADRetrieve) -> Result<RadonTypes> {
     // Disable all execution tracing features, as this is the best-effort version of this method
-    run_retrieval_report(retrieve, RadonScriptExecutionSettings::disable_all())
+    run_retrieval_report(retrieve, RadonScriptExecutionSettings::disable_all(), None)
         .await
         .map(RadonReport::into_inner)
 }
 
+/// Run retrieval using multiple transports, and only produce a positive result if the retrieved
+/// values pass the filter function from the tally stage.
+///
+/// The idea behind this is to avoid producing commitments for data requests with sources that act
+/// in an inconsistent way, i.e. they return very different values when queried through different
+/// HTTP transports at once.
+pub async fn run_paranoid_retrieval(
```

Comment:
Love the name xD

Reply:
Just by doing a couple of tests it can be seen that several proxies are not able to access some APIs. Shall we consider that timeout error as a valid input for the filter? In my opinion we should discard them. Of course in that case we should discuss what to do if all cases gave timeout